src/main/java/gov/nasa/cumulus/CNMResponse.java

```diff
@@ -126,6 +139,18 @@ public static String generateOutput(String inputCnm, String exception, JsonObjec
 		return new Gson().toJson(inputKey);
 	}
 
+	/**
+	 * Parses S3 bucket and key based on regex.
+	 *
+	 * @param s3Path path to archived location of file
+	 * @return Matcher object where group(1) is the bucket and group(2) is the key
+	 */
+	public static Matcher getSourceBucketAndKey(String s3Path) {
+		Pattern p = Pattern.compile("s3://([^/]*)/(.*)");
+		Matcher m = p.matcher(s3Path);
+		return m;
+	}
+
     public String getError(JsonObject input, String key){
```

Comment:
When testing I noticed error type log messages that were false alarm. If there is no error input.get(key) is None string. Should update AdapterLogger.LogError to AdapterLogger.LogDebug in this function.

```
{
    "executions": "dc59e4e8-93d1-4805-8f68-84a6a201483e",
    "sender": "podaac-sndbx-cumulus-CnmResponse",
    "level": "error",
    "granules": "[\"SWOT_IVK_20210612T081400_20210612T072103_20210612T080137_O_APID1075.PTM_1\"]",
    "stackName": "podaac-sndbx-cumulus",
    "message": "gov.nasa.cumulus.CNMResponse WorkflowException:\"None\"",
    "version": "$LATEST",
    "timestamp": "2020-11-04T23:17:11.193Z"
}
```